apiserver/admin_test.go

```diff
@@ -804,13 +787,49 @@ func (s *loginSuite) TestOtherEnvironmentWhenNotController(c *gc.C) {
 	defer envState.Close()
 	info.ModelTag = envState.ModelTag()
 	st := s.openAPIWithoutLogin(c, info)
-	defer st.Close()
 
 	err := st.Login(machine.Tag(), password, "nonce", nil)
-	c.Assert(errors.Cause(err), gc.DeepEquals, &rpc.RequestError{
-		Message: "permission denied",
-		Code:    "unauthorized access",
+	assertPermissionDenied(c, err)
+}
+
+func (s *loginSuite) loginUserWithAccess(c *gc.C, info *api.Info) (*state.User, params.LoginResult) {
+	password := "shhh..."
+	user := s.Factory.MakeUser(c, &factory.UserParams{
+		Password: password,
 	})
+	conn := s.openAPIWithoutLogin(c, info)
+
+	var result params.LoginResult
+	request := &params.LoginRequest{
+		AuthTag:     user.Tag().String(),
+		Credentials: password,
+	}
+	err := conn.APICall("Admin", 3, "", "Login", request, &result)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result.UserInfo, gc.NotNil)
+	return user, result
+}
+
+func (s *loginSuite) TestLoginResultModelUser(c *gc.C) {
+	info, cleanup := s.setupServer(c)
+	defer cleanup()
+
+	user, result := s.loginUserWithAccess(c, info)
+	c.Check(result.UserInfo.Identity, gc.Equals, user.Tag().String())
+	c.Check(result.UserInfo.ControllerAccess, gc.Equals, "login")
+	c.Check(result.UserInfo.ModelAccess, gc.Equals, "admin")
+}
+
+func (s *loginSuite) TestLoginResultModelUserEveryoneCreateOnlyNonLocal(c *gc.C) {
+	info, cleanup := s.setupServer(c)
+	defer cleanup()
+
+	setEveryoneAccess(c, s.State, s.AdminUserTag(c), permission.AddModelAccess)
```

Comment:
Mentioned in another comment, this is the test that appears to check for incorrect behaviour. We are granting addmodel to everyone, but below only checking for "login"